code/modules/mob/living/basic/space_fauna/hivebot/hivebot_behavior.dm

```diff
@@ -0,0 +1,72 @@
+/datum/ai_behavior/find_and_set/hive_partner
+
+/datum/ai_behavior/find_and_set/hive_partner/search_tactic(datum/ai_controller/controller, locate_path, search_range)
+	var/mob/living/living_pawn = controller.pawn
+	var/list/hive_partners = list()
+	for(var/mob/living/target in oview(10, living_pawn))
+		if(!istype(target, locate_path))
+			continue
+		if(target.stat == DEAD)
+			continue
+		hive_partners += target
+
+	if(length(hive_partners))
+		return pick(hive_partners)
+
+
+/datum/ai_behavior/relay_message
+	var/length_of_message = 4
+	behavior_flags = AI_BEHAVIOR_REQUIRE_MOVEMENT|AI_BEHAVIOR_REQUIRE_REACH
```

Comment:
`AI_BEHAVIOR_CAN_PLAN_DURING_EXECUTION` may be useful here, it means that if the bot could acquire an alternate target (a hostile has come into view?) it may immediately abandon this lowest priority action.